test/services/link/index.test.js

```diff
@@ -26,8 +80,124 @@ describe('link service', () => {
+						if (error) return reject(error);
+						chai.expect(result.status).to.equal(200);
+						chai.expect(result.body).to.include.keys('target');
+						return resolve();
+					});
+			}));
+	});
+
+	it('generates an external link from 31 days ago', function () {
+		this.timeout(10000);
+		const createdAt = subtractDays(new Date(), 31);
+
+		const url = 'https://google.de/registration/';
+		return service.create({ target: url, createdAt, forceNew })
+			.then((data) => {
+				chai.expect(data.id).to.have.lengthOf(service.Model.linkLength);
+				chai.expect(data.target).to.equal(url);
+				return Promise.resolve(data.id);
+			})
+			.then((id) => new Promise((resolve, reject) => {
+				chai.request(app)
+					.get(`/link/${id}`)
+					.end((error, result) => {
+						if (error) return reject(error);
+						chai.expect(result.status).to.equal(200);
+						chai.expect(result.body).to.include.keys('target');
+						return resolve();
+					});
+			}));
+	});
+
+	it('generates a valid link two days in the future', function () {
+		this.timeout(10000);
+		const createdAt = subtractDays(new Date(), -2);
+
+		const url = 'http://localhost:3100/registration/';
+		return service.create({ target: url, createdAt, forceNew })
+			.then((data) => {
+				chai.expect(data.id).to.have.lengthOf(service.Model.linkLength);
+				chai.expect(data.target).to.equal(url);
+				return Promise.resolve(data.id);
+			})
+			.then((id) => new Promise((resolve, reject) => {
+				chai.request(app)
+					.get(`/link/${id}`)
+					.end((error, result) => {
+						if (error) return reject(error);
+						chai.expect(result.status).to.equal(200);
+						chai.expect(result.body).to.include.keys('target');
+						return resolve();
+					});
+			}));
+	});
+
+	it('generates a valid link from now', function () {
+		this.timeout(10000);
+		const createdAt = new Date();
+
+		const url = 'http://localhost:3100/registration/';
+		return service.create({ target: url, createdAt, forceNew })
+			.then((data) => {
+				chai.expect(data.id).to.have.lengthOf(service.Model.linkLength);
+				chai.expect(data.target).to.equal(url);
+				return Promise.resolve(data.id);
+			})
+			.then((id) => new Promise((resolve, reject) => {
+				chai.request(app)
+					.get(`/link/${id}`)
+					.end((error, result) => {
+						if (error) return reject(error);
+						chai.expect(result.status).to.equal(200);
+						chai.expect(result.body).to.include.keys('target');
+						return resolve();
+					});
+			}));
+	});
+
+	it('generate two links with same target', function () {
+		this.timeout(10000);
+		const createdAt = new Date();
+
+		const url = 'http://localhost:3100/testurl/1234';
+		return service.create({ target: url, createdAt })
+			.then((data) => {
+				console.log(data._id)
+				chai.expect(data.id).to.have.lengthOf(service.Model.linkLength);
+				chai.expect(data.target).to.equal(url);
+				return Promise.resolve(data.id);
+			})
+			.then((id) => new Promise((resolve, reject) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['reject' is defined but never used. Allowed unused args must match /app|req|res|next|options|params/u.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=3811219)